src/ui/groups_widgets.cc

```diff
@@ -790,34 +790,73 @@ void DictGroupsWidget::addAutoGroupsByFolders()
     return;
   }
 
-  QMultiMap< QString, sptr< Dictionary::Class > > dictMap;
+  // Map from dict to ContainerFolder's parent
+  QMap< sptr< Dictionary::Class >, QString > dictToContainerFolder;
 
   for ( const auto & dict : *activeDicts ) {
-    QString dictFolder = dict->getContainingFolder();
+    QDir c = dict->getContainingFolder();
+    c.cdUp();
+    dictToContainerFolder.insert( dict, c.absolutePath() );
+  }
 
-    if ( dictFolder.isEmpty() ) {
-      continue;
+  /*
+  Check for duplicated ContainerFolder's parent folder names, and prepend the upper parent's name.
+
+  .
+  ├─epistularum
+  │   └─Japanese   <- Group
+  │       └─DictA  <- Dict Files's container folder
+  |          └─ DictA Files
+  ├─Mastameta
+  │   └─Japanese   <- Group
+  |       └─DictB  <- Dict Files's container folder
+  |          └─ DictB Files
+
+  will be grouped into epistularum/Japanese and Mastameta/Japanese
+  */
+
+  // set a dirname to true if there are duplications like `epistularum/Japanese` and `Mastameta/Japanese`
+  QHash< QString, bool > dirNeedPrepend{};
+
+  for ( const auto & path : dictToContainerFolder.values() ) {
+    auto dir = QDir( path );
+    if ( dirNeedPrepend.contains( dir.dirName() ) ) {
+      dirNeedPrepend[ dir.dirName() ] = true;
+    }
+    else {
+      dirNeedPrepend.insert( dir.dirName(), false );
     }
+  }
```

Comment:
If a user knows this feature, and deliberately create conflicting folder/group names, then wcyd `:)`.

We can add existing group names, though.

Reply:
No we cannot. If we check existing group names, then it will always conflict with previous "Auto group by folder" groups.

Click "Auto group by folder" multiple times is possible.

We should just overwrite existing ones if a group name is equal to "Auto group by folder"'s result group name.

Reply:
this is a case